src/main/java/org/folio/dao/invoice/InvoicePostgresDAO.java

```diff
@@ -101,6 +70,14 @@ public Future<DBClient> createInvoice(Invoice invoice, DBClient client) {
     return promise.future();
   }
 
+  @Override
+  public Future<Void> updateInvoice(Invoice invoice, Conn conn) {
+    return conn.update(INVOICE_TABLE, invoice, invoice.getId())
+      .onFailure(t -> log.error("updateInvoice failed for invoice with id {}", invoice.getId(), t))
```

Comment:
`updateInvoice` will always return succeded future in this case. Possible exception occured while updating invoice will be just logged and ignored. Is this expected behavior?

Reply:
I don't understand. `onFailure` does not prevent a failed future from being returned (like `recover`), it is just an event handler for when it fails...

Reply:
right. sorry for confusing